### PR TITLE
[3.10] Backport the image Layout

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes

This PR backports the `image.php` JLayout so Devs could use it even if their extensions are targeting both 3.10 and 4.x

### Testing Instructions

in the protostar index.php add before the `<Jdoc:include type="component"` the following

```php

<?php echo JLayoutHelper::render(
 'joomla.html.image',
 [
 	'src' => JUri:root() . 'images/joomla_black.png',
 	'alt' => 'Some Alt tag',
 ]
); ?>

<?php echo JLayoutHelper::render(
 'joomla.html.image',
 [
 	'src' => JUri:root() . 'iimages/powered_by.png',
 	'alt' => false,
 ]
); ?>
```

You should be able to see the 2 Joomla banners, one with alt: Some Alt tag and the other one without any alt tag

### Documentation Changes Required

Yes, this needs to be communicated, eg: this is the preferred way to handle images.
I'm preparing something...

@zero-24 this is a backport of #35780

### CODE REVIEW

File: layouts/joomla/html/image.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * @package     Joomla.Site
+ * @subpackage  Layout
+ *
+ * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
```
Comment (brianteeman) [inline layouts/joomla/html/image.php]:
```suggestion
 * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
```

happy new year

File: layouts/joomla/html/image.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Layout variables
+ * -----------------
+ * @var   array  $displayData  Array with all the given attributes for the image element.
+ *                             Eg: src, class, alt, width, height, loading, decoding, style, data-*
+ *                             Note: only the alt and src attributes are escaped by default!
+ */
+defined('_JEXEC') or die;
+
+if (isset($displayData['alt']))
```
Comment (Quy) [inline layouts/joomla/html/image.php]:
```suggestion
if (isset($displayData['src']))
```

Comment (dgrammatiko) [reply layouts/joomla/html/image.php]: 5 lines of code 20 errors...

File: layouts/joomla/html/image.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * Layout variables
+ * -----------------
+ * @var   array  $displayData  Array with all the given attributes for the image element.
+ *                             Eg: src, class, alt, width, height, loading, decoding, style, data-*
+ *                             Note: only the alt and src attributes are escaped by default!
+ */
+defined('_JEXEC') or die;
+
```
Comment (Quy) [inline layouts/joomla/html/image.php:18]: This is missing?

`$img = HTMLHelper::_('cleanImageURL', $displayData['src']);`

Comment (dgrammatiko) [reply layouts/joomla/html/image.php:18]: It doesn’t exist in 3.x

File: layouts/joomla/html/image.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+/**
+ * Layout variables
+ * -----------------
+ * @var   array  $displayData  Array with all the given attributes for the image element.
+ *                             Eg: src, class, alt, width, height, loading, decoding, style, data-*
+ *                             Note: only the alt and src attributes are escaped by default!
+ */
+defined('_JEXEC') or die;
```
Comment (PhilETaylor) [inline layouts/joomla/html/image.php]: this defined  should come after the opening file header and before the layout vars docs.

**Comment by laoneo** [comment]
Getting the following warnings:
Warning: Undefined variable $img in /var/www/html/pr/36613/layouts/joomla/html/image.php on line 21
Warning: Attempt to read property "url" on null in /var/www/html/pr/36613/layouts/joomla/html/image.php on line 21

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on f05a9722f2fdf5326c7a498803bfc769984a184c<br /><br />test code is wrong, after i fixed, it worked<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36613">issues.joomla.org/tracker/joomla-cms/36613</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on f05a9722f2fdf5326c7a498803bfc769984a184c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36613">issues.joomla.org/tracker/joomla-cms/36613</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36613">issues.joomla.org/tracker/joomla-cms/36613</a>.</sub>

**Comment by zero-24** [comment]
Merging here thanks :)

**Pull request merged**